kv/replica_slice.go

```diff
@@ -103,6 +106,8 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 		return 0
 	}
 	topIndex := len(rs) - 1
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
You shouldn't need to seed on every call.
We'll have to see whether it's ok that you're using the global `rand` instance here; for testing purposes it could be nicer if you pass a `rand.Rand` in with the call to `SortByCommonAttributePrefix` (which would likely be an instance sitting on the `DistSender`).